miden-tx/tests/integration/scripts/faucet.rs

```diff
@@ -97,16 +54,20 @@ fn prove_faucet_contract_mint_fungible_asset_succeeds() {
         note_execution_hint = Felt::from(note_execution_hint)
     );
 
-    let tx_args = build_tx_args_from_script(&tx_script_code);
+    let tx_script =
+        TransactionScript::compile(tx_script_code, vec![], TransactionKernel::testing_assembler())
+            .unwrap();
+    let tx_context = mock_chain
+        .build_tx_context(faucet.account().id(), &[], &[])
+        .tx_script(tx_script)
+        .build();
 
-    let executed_transaction = executor
-        .execute_transaction(faucet_account.id(), block_ref, &note_ids, tx_args)
-        .unwrap();
+    let executed_transaction = tx_context.execute().unwrap();
 
-    prove_and_verify_transaction(executed_transaction.clone()).unwrap();
+    assert!(prove_and_verify_transaction(executed_transaction.clone()).is_ok());
```

Comment:
Nit: I would prefer `result.unwrap()` over `assert!(result.is_ok())` because they both serve as an assertion but the former will print the error if it fails, while the latter only says something like "expected true, found false" which is less helpful for debugging.